docs/platform-specification.md

```diff
@@ -625,6 +625,64 @@ The tool configuration properties are available globally without the prefix. For
 property can be used as **{cmd.path}** inside the recipe, and the same happens for all the other avrdude configuration
 variables.
 
+#### Pluggable Discovery
+
+Discovery tools are a special kind of tools that are used to find supported boards, a platform must declare one or more
+Pluggable Discoveries in its `platform.txt`. It's ok to use another platform discovery or the builtin ones if necessary.
+
+There are two different syntaxes to declare discoveries, if the platform uses just one discovery:
+
+```
+discovery.required=PLATFORM:DISCOVERY_NAME
+```
+
+instead if it needs multiple discoveries:
+
+```
+discovery.required.0=PLATFORM:DISCOVERY_ID_1
+discovery.required.1=PLATFORM:DISCOVERY_ID_2
+```
+
+A platform that supports only boards connected to the serial can easily use the builtin `serial-discovery` without
+creating a custom Pluggable Discovery:
+
+```
+discovery.required=builtin:serial-discovery
+```
+
+if it supports also boards connected to the network it can use the builtin `mdns-discovery`:
+
+```
+discovery.required.0=builtin:serial-discovery
+discovery.required.1=builtin:mdns-discovery
+```
+
+Since the above syntax requires adding a discovery in the `discoveryDependencies` field in the platform
+`package_index.json` it might be cumbersome to do it before releasing it so we also provide another syntax to ease
+development:
```

Comment:
It should be possible to keep both with different IDs but they would both be started by the CLI if they're in the `platform.txt`, either one must be omitted or commented out.

Reply:
OK. I was thinking that if there was a way for them to coexist then we might want to find a way to explain that, since I foresee this will be something of interest to platform developers. Since there isn't, I think the system is explained well enough.